tests/testthat/test-chrome.R

```diff
@@ -96,19 +96,87 @@ test_that("once closed, is_alive() return FALSE", {
 })
 
 
-# chrome_execute ----------------------------------------------------------
+context("test chrome_execute")
 
 test_that("With only 1 argument chrome_execute() returns the value of the async function invisibly", {
   value <- runif(1)
   async_fun <- function(client) {
     promises::promise_resolve(value)
   }
-  expect_equal(chrome_execute(async_fun), value)
-  expect_invisible(chrome_execute(async_fun))
+  expect_identical(chrome_execute(async_fun), value)
+  # if not async, return invisibly
+  expect_invisible(chrome_execute(async_fun, async = FALSE))
+  # if async, result is a promise
+  pr <- chrome_execute(async_fun, async = TRUE)
+  expect_s3_class(pr, "promise")
+  hold(pr)
 })
 
+
 test_that("With multiple argument, chrome_execute() return a list with the values of the async functions", {
   values <- as.list(runif(3))
-  async_funs <- values %>% purrr::map(~ function(client) promises::promise_resolve(.x))
+  async_funs <- purrr::map(values, ~ function(client) promises::promise_resolve(.x))
   expect_identical(do.call(chrome_execute, async_funs), values)
 })
+
+test_that("rejects if one promises get rejected", {
+  async_funs <- list(fun1 = function(client) promises::promise_resolve(1L),
+                     fun2 = function(client) promises::promise_reject("fun2 rejected"))
+  expect_error(chrome_execute(.list = async_funs), "fun2 rejected")
+})
+
+test_that("timeouts are respected and recycled (using JS in chrome)", {
+  async_fun <- function(ms) {
+    function(client) {
+      Runtime <- client$Runtime
+      Runtime$enable() %...>% {
+        Runtime$evaluate(
+          expression = paste0(
+            'function sleep(ms) {',
+            'return new Promise(resolve => setTimeout(resolve, ms));',
+            '};',
+            'async function wait() {',
+            'var n = Date.now();',
+            'await sleep(', ms, ');',
+            'var n2 = Date.now()-n;',
+            'console.log(n2); return n2;}',
+            'wait();'),
+          awaitPromise = TRUE
+        )}
+    }
+  }
+  async_funs <- list(
+    async_fun(500),
+    async_fun(3000)
+  )
+  expect_error(chrome_execute(.list = async_funs, timeouts = c(30, 1)),
+               "The delay of 1 seconds expired in async function n-2.")
+  expect_error(chrome_execute(.list = async_funs, timeouts = c(2)),
+               "The delay of 2 seconds expired in async function n-2.")
+  skip_on_os("windows")
+  expect_error(chrome_execute(.list = async_funs, timeouts = c(1)),
+               "The delay of 1 seconds expired in async function n-2.")
```

Comment:
I ran this test on win 7. 
I get this error message:
```
Error: The delay of 1 seconds expired in async function n-1.
```
I also get these warnings:
```
[2019-05-06 15:03:23] [info] asio async_connect error: asio.system:10061 (Aucune connexion n’a pu être établie car l’ordinateur cible l’a expressément refusée.)
[2019-05-06 15:03:23] [info] Error getting remote endpoint: asio.system:10057 (Une requête d’envoi ou de réception de données n’a pas été autorisée car le socket n’est pas connecté et (lors de l’envoi sur un socket datagramme en utilisant un appel sendto) aucune adresse n’a été fournie.)
[2019-05-06 15:03:23] [error] handle_connect error: Aucune connexion n’a pu être établie car l’ordinateur cible l’a expressément refusée.
Warning: Aucune connexion n’a pu être établie car l’ordinateur cible l’a expressément refusée.
```

Reply:
These are the same warnings !